go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -1006,7 +1008,7 @@ func (be *BuiltinBackupEngine) restoreFiles(ctx context.Context, params RestoreP
 	}
 
 	if bm.Incremental {
-		createdDir, err = os.MkdirTemp("", "restore-incremental-*")
+		createdDir, err = os.MkdirTemp(vtenv.VtDataRoot(), "restore-incremental-*")
```

Comment:
Do we copy only the binary logs here or the full backup? If it's the full backup, this can have a significant effect on the storage needed and we might want to make this configurable (I'm assuming that the vtdataroot has *another* copy of the actual data that was restored prior to this step?). `/tmp` may also be direct attached storage or `tmpfs` which could cause a significant performance impact. It may make sense for this to be a flag with the default value being `/tmp`.

Reply:
To answer some of my own questions... 🙂 

https://vitess.io/docs/user-guides/operating-vitess/backup-and-restore/creating-a-backup/#create-an-incremental-backup-with-vtctl

So we’re restoring an incremental backup here, which was only binary logs to begin with. And incremental is always done by the builtinengine, regardless of which engine you’re using for full backups.